zeppelin-interpreter/src/main/java/org/apache/zeppelin/conf/ZeppelinConfiguration.java

```diff
@@ -119,6 +121,25 @@ public static synchronized ZeppelinConfiguration create() {
       url = classLoader.getResource(ZEPPELIN_SITE_XML);
     }
 
+    // Support for using ZeppelinConfiguration in RemoteInterpreterServer
+    if (url == null) {
```

Comment:
Why add this piece of code ? What happens without it ?

Reply:
Because `RemoteInterpreterServer.java` needs to get the configuration information of zconf.
Obtain the configuration parameters of the cluster and connect to the cluster.
```
<property>
  <name>zeppelin.cluster.addr</name>
  <value>10.120.196.234:6000,10.120.196.235:6000,10.120.196.236:6000</value>
</property>
```

Reply:
Sorry, I didn't get it. Why RemoteInterpreterServer could not get `zeppelin-site.xml` ？ It should be launched in any host of cluster, and ZEPPELIN_HOME, ZEPPELIN_CONF_DIR should be passed to interpreter process properly.

Reply:
Ok, I modified to get the `zeppelin-site.xml` path from the environment variable `ZEPPELIN_HOME` or `ZEPPELIN_CONF_DIR `.